pkg/api/staking.go

```diff
@@ -0,0 +1,119 @@
+package api
+
+import (
+	"encoding/hex"
+	"errors"
+	"math/big"
+	"net/http"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/staking/stakingcontract"
+	"github.com/gorilla/mux"
+)
+
+func (s *Service) stakingAccessHandler(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !s.stakingSem.TryAcquire(1) {
+			s.logger.Debug("staking access: simultaneous on-chain operations not supported")
+			s.logger.Error(nil, "staking access: simultaneous on-chain operations not supported")
+			jsonhttp.TooManyRequests(w, "simultaneous on-chain operations not supported")
+			return
+		}
+		defer s.stakingSem.Release(1)
+
+		h.ServeHTTP(w, r)
+	})
+}
+
+type getStakeResponse struct {
+	StakedAmount *big.Int `json:"stakedAmount"`
+}
+
+func (s *Service) stakingDepositHandler(w http.ResponseWriter, r *http.Request) {
+	overlayAddr, err := hex.DecodeString(mux.Vars(r)["address"])
+	if err != nil {
+		s.logger.Debug("get stake: decode overlay string failed", "string", overlayAddr, "error", err)
```

Comment:
is ` "string"` filed name intentionally chosen? shouldn't it be `"address"` or `"overlay"` ?